src/script/client/ClientEntity.ts

```diff
@@ -53,62 +62,56 @@ class ClientEntity {
       isVerified: ko.observable(false),
       primaryKey: undefined,
     };
-
-    this.session = {};
   }
 
   /**
    * Splits an ID into user ID & client ID.
-   * @param {string} id - Client ID to be dismantled
-   * @returns {Object} Object containing the user ID & client ID
    */
-  static dismantleUserClientId(id) {
-    const [userId, clientId] = _.isString(id) ? id.split('@') : [];
+  static dismantleUserClientId(id: string): {clientId: string; userId: string} {
+    const [userId, clientId] = isString(id) ? id.split('@') : ([] as string[]);
     return {clientId, userId};
   }
 
   /**
    * Returns the ID of a client in a format suitable for UI display in user preferences.
-   * @returns {Array<string>} Client ID in pairs of two as an array
+   * @returns Client ID in pairs of two as an array
    */
-  formatId() {
+  formatId(): string[] {
     return zeroPadding(this.id, 16).match(/.{1,2}/g);
   }
 
   /**
-   * @returns {boolean} True, if the client is the self user's permanent client.
+   * @returns `true`, if the client is the self user's permanent client.
    */
-  isPermanent() {
+  isPermanent(): boolean {
     return this.type === ClientType.PERMANENT;
   }
 
   /**
-   * @returns {boolean} - True, if it is NOT the client of the self user.
+   * @returns `true`, if it is **not** the client of the self user.
```

Comment:
We should check for `this.isSelfClient` if we want to be sure that it only affects self clients.